### PR TITLE
Update Contributing Docs

### PR DESCRIPTION
**Pull request by imurchie**
## Proposed changes

First pass at updating the contributing documentation. I am sure it will take more iterations to get it into a good state.

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/en/contributing-to-appium/appium-from-source.md [hunk 1]
```diff
@@ -80,36 +75,37 @@ have to modify your `/etc/authorization` file in one of two ways:
...
 	```
 
 At this point, run:
 
-```center
-rm -rf node-modules
-npm install
-gulp transpile
+```
+rm -rf node-modules && npm install
```
Comment (jlipps) [inline docs/en/contributing-to-appium/appium-from-source.md]: should we include package-lock.json in the rm?

File: docs/en/contributing-to-appium/style-guide.md [hunk 2]
```diff
@@ -2,8 +2,16 @@
...
 writing javascript. Please conform to these so we can merge your pull request
- without going back and forth about style. The main principle is: *make your
- code look like the surrounding code*.
+without going back and forth about style. The main principle is: *make your
+code look like the surrounding code*.
+
+### JavaScript
+
+With the exception of the code that runs on devices ([appium-uiautomator2-server](https://github.com/appium/appium-uiautomator2-server) for 
+Android, [WebDriverAgent](https://github.com/appium/WebDriverAgent) for iOS), Appium is written in [Node.js](). If you are
```
Comment (jlipps) [inline docs/en/contributing-to-appium/style-guide.md]: url for node.js link?

**Review by jlipps — COMMENTED** [review]
looks good!

File: docs/en/contributing-to-appium/appium-from-source.md (hunk 1 above)
Comment (imurchie) [reply docs/en/contributing-to-appium/appium-from-source.md]: Good point. I'll add a little blurb. I think _most_ repos now don't produce them, but I may have missed some.

File: docs/en/contributing-to-appium/style-guide.md (hunk 2 above)
Comment (imurchie) [reply docs/en/contributing-to-appium/style-guide.md]: Good catch. I originally did this on an airplane with no interwebs, so had to leave the links blank and go back and fill in. Obviously I missed some. :(

File: docs/en/contributing-to-appium/appium-from-source.md (hunk 1 above)
Comment (imurchie) [reply docs/en/contributing-to-appium/appium-from-source.md]: I also appear to have used `node-modules` instead of `node_modules`. So doubly good catch!

**Pull request merged**